File: SymmetricTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        // Calling isSymmetric Overloaded Function
        return isSymmetric(root, root);
    }
    
    // Helper Function
    public boolean isSymmetric(TreeNode t1, TreeNode t2){
        
        // If All Conditions Satisfy And Both Nodes Are Null Then Return True
        if(t1 == null && t2 == null){
            return true;
        }
        
        // If All Conditions Satisfy But One Reaches Null Then Return False
        if(t1 == null || t2 == null){
            return false;
        }
        
        // Calling Function With Left Right And Right Left Combo And Also Checking For Values
        return t1.val == t2.val && isSymmetric(t1.left, t2.right) && isSymmetric(t1.right, t2.left);
    }
}

File: SymmetricTree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function isSymmetric(root: TreeNode | null): boolean {
    const checkSymmetry = (p:TreeNode, q:TreeNode) : boolean => {
        if(!p && !q) return true;
        if(!p || !q || p.val !== q.val) return false;
        return checkSymmetry(p.left, q.right) && checkSymmetry(p.right, q.left);
    }
    return checkSymmetry(root.left, root.right);
};
